conan/tools/env/environment.py

```diff
@@ -520,12 +520,52 @@ def save_sh(self, file_location, generate_deactivate=True):
         content = f'script_folder="{os.path.abspath(filepath)}"\n' + content
         save(file_location, content)
 
+    def save_fish(self, file_location, generate_deactivate=True):
+        filepath, filename = os.path.split(file_location)
+        deactivate_file = os.path.join(filepath, "deactivate_{}".format(filename))
+        values = self._values.keys()
+        if len(values) == 0:
+            # Empty environment, nothing to restore (Easier to handle in Fish)
+            deactivate = ("""echo "echo Nothing to restore" > {deactivate_file}"""
+                          .format(deactivate_file=deactivate_file))
+        else:
+            deactivate = textwrap.dedent("""\
```

Comment:
For new one, I'd try to do directly an in-memory or in-tmp-folder deactivate script rather than a local generated one.